rust/arrow/src/datatypes.rs

```diff
@@ -22,6 +22,7 @@
 //!  * [`Field`](crate::datatypes::Field) to describe one field within a schema.
 //!  * [`DataType`](crate::datatypes::DataType) to describe the type of a field.
 
+use std::collections::BTreeMap;
```

Comment:
Is there value in using `BTreeMap` for both metadata types?

Reply:
Copied from [rust doc std/collections](https://doc.rust-lang.org/std/collections/index.html):

**`Use a HashMap when:**

You want to associate arbitrary keys with an arbitrary value.
You want a cache.
You want a map, with no extra functionality.

**Use a BTreeMap when:**

You want a map sorted by its keys.
You want to be able to get a range of entries on-demand.
You're interested in what the smallest or largest key-value pair is.
You want to find the largest or smallest key that is smaller or larger than something.`

The only reason that use `BTreeMap` for `Field` is: don't break the `'Hash`, `PartialOrd`, `Ord`.
I almost can't image any situation that will sort schemas, unless compare/sort a schema by versions.